pytermgui/ansi_interface.py

```diff
@@ -284,12 +301,35 @@ def fill(self, color: int = 0, flush: bool = True) -> None:
         if flush:
             sys.stdout.flush()
 
+    @contextmanager
+    def no_echo(self) -> None:
+        """Context manager which disables Echo mode and restores it after closing"""
+
+        if _IS_NT:
+            # WARNING! EXPERIMENTAL CODE
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-10), 0)
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), 7)
+        else:
+            tmp = self._original_stdin
+            tmp[3] = tmp[3] & ~(termios.ECHO | termios.ICANON)
+            termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, tmp)
+        try:
+            yield
+        finally:
+            if _IS_NT:
+                kernel32.SetConsoleMode(
+                    kernel32.GetStdHandle(-10), self._original_stdin
+                )
+                kernel32.SetConsoleMode(
+                    kernel32.GetStdHandle(-11), self._original_stdout
+                )
+            else:
+                termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, self._original_stdin)
```

Comment:
This content would be better suited inside `set_echo` & `unset_echo`, and a context manager like this should only call those methods.

Reply:
After I will refactor those 2 methods I will change it.
This PR is only for adding this context manager. (And few small fixes which I already described)

Reply:
I'd say adding better `echo` functions would fit into this PR, and would avoid making another one.

Reply:
Alright. I will change it

So this PR will modify currently existing **echo** switches and also add **echo** context manager to `_Terminal` class.

I will do it in a few hours since I am busy right now alright?